berkeley-db.m4: Avoid bdb version detection; just link to libaprutil instead

In Debian, apu-config --libs does not print dependent libraries (see
#378105).  svn relies on this to detect the BDB version.  Work around
this by linking to -laprutil1 instead, which will indirectly link to
libdb.

I don't think this can be sent upstream, as it won't work when
compiling aprutil in-tree.  But I'm not certain.

--- a/build/ac-macros/berkeley-db.m4
+++ b/build/ac-macros/berkeley-db.m4
@@ -183,7 +183,7 @@
     SVN_DB_LIBS=["${SVN_DB_LIBS-`$apu_config --libs | $SED -e 's/.*\(-ldb[^[:space:]]*\).*/\1/' | $EGREP -- '-ldb[^[:space:]]*'`}"]
 
     CPPFLAGS="$SVN_DB_INCLUDES $SVN_APRUTIL_INCLUDES $CPPFLAGS" 
-    LIBS="`$apu_config --ldflags` $SVN_DB_LIBS $LIBS"
+    LIBS="`$apu_config --ldflags --link-ld` $SVN_DB_LIBS $LIBS"
 
     if test -n "$svn_berkeley_db_header"; then
       SVN_DB_HEADER="#include <$svn_berkeley_db_header>"
